web/src/engine/websites/Luscious.ts

```diff
@@ -0,0 +1,123 @@
+
+type APIMangaPage = {
+    data: {
+        album: {
+            list: {
+                items: APIManga[]
+            }
+        }
+    }
+};
+
+type APIManga = {
+    id: number,
+    title: string,
+    slug: string,
+    url: string
+};
+
+type PagesResult = {
+    data: {
+        picture: {
+            list: {
+                info: {
+                    has_next_page: boolean
+                },
+                items: { url_to_original: string }[]
+            }
+        }
+    }
+}
+
+@Common.ChaptersUniqueFromManga()
+@Common.ImageAjax()
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('luscious', `Luscious`, 'https://www.luscious.net', Tags.Language.Multilingual, Tags.Source.Aggregator, Tags.Rating.Pornographic);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return /https?:\/\/www\.luscious\.net\/albums\//.test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const uri = new URL(url);
+        const id = uri.pathname.match(/_(\d+)\/?$/)[1];
+        const request = new FetchRequest(url);
+        const name = (await FetchCSS(request, 'main h1.album-heading')).pop().textContent.trim();
+        return new Manga(this, provider, id, name);
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangaList = [];
+        for (let page = 1, run = true; run; page++) {
+            const mangas = await this._getMangasFromPage(page, provider);
+            mangas.length > 0 ? mangaList.push(...mangas) : run = false;
+        }
+        return mangaList;
+    }
+
+    async _getMangasFromPage(page: number, provider: MangaPlugin): Promise<Manga[]> {
+        const url = new URL(apiUrl);
+        url.searchParams.set('operationName', 'AlbumList');
+        url.searchParams.set('query', '%20query%20AlbumList(%24input%3A%20AlbumListInput!)%20%7B%20album%20%7B%20list(input%3A%20%24input)%20%7B%20info%20%7B%20...FacetCollectionInfo%20%7D%20items%20%7B%20...AlbumInSearchList%20%7D%20%7D%20%7D%20%7D%20fragment%20FacetCollectionInfo%20on%20FacetCollectionInfo%20%7B%20page%20has_next_page%20has_previous_page%20total_items%20total_pages%20items_per_page%20url_complete%20%7D%20fragment%20AlbumInSearchList%20on%20Album%20%7B%20__typename%20id%20title%20description%20created%20modified%20like_status%20moderation_status%20number_of_favorites%20number_of_dislikes%20number_of_pictures%20number_of_animated_pictures%20number_of_duplicates%20slug%20is_manga%20url%20download_url%20labels%20permissions%20cover%20%7B%20width%20height%20size%20url%20%7D%20created_by%20%7B%20id%20url%20name%20display_name%20user_title%20avatar_url%20%7D%20language%20%7B%20id%20title%20url%20%7D%20tags%20%7B%20category%20text%20url%20count%20%7D%20genres%20%7B%20id%20title%20slug%20url%20%7D%20%7D%20');
```

Comment:
hard to read, use `' query AlbumList($input: AlbumListInput!) { album { list(input: $input) { info { ...FacetCollectionInfo } items { ...AlbumInSearchList } } } } fragment FacetCollectionInfo on FacetCollectionInfo { page has_next_page has_previous_page total_items total_pages items_per_page url_complete } fragment AlbumInSearchList on Album { __typename id title description created modified like_status moderation_status number_of_favorites number_of_dislikes number_of_pictures number_of_animated_pictures number_of_duplicates slug is_manga url download_url labels permissions cover { width height size url } created_by { id url name display_name user_title avatar_url } language { id title url } tags { category text url count } genres { id title slug url } } '` instead

Reply:
There are also some fields which probably can be removed as well

Reply:
Other queries in this file as well

Reply:
About the "hard to read part" , this is sadly on purpose. I had a hard time to make it work because i messed with the encoding. This website use GET for the queries and so they tend to over-encode everything.  I'll try to see it it works 🤷 .

Clearly using POST would be better, they are crazy :D